src/main/java/org/candlepin/service/CloudRegistrationAdapter.java

```diff
@@ -93,4 +94,28 @@ CloudAccountData setupCloudAccountOrg(String cloudAccountID, String cloudOfferin
         CloudProvider cloudProviderShortName, String ownerKey)
         throws CouldNotAcquireCloudAccountLockException, CouldNotEntitleOrganizationException;
 
+    /**
+     * Create (if one does not already exist) and entitle an owner upstream of Candlepin, for the given
+     * cloud account id and cloud offering id.
+     *
+     * @param cloudAccountID
+     *     Identifier for cloud account that needs access to Red Hat content
+     *
+     * @param cloudProviderShortName
+     *     Shortcut of the cloud provider from which the offering came
+     *
+     * @param skuID
+     *     Product ID from Anonymous Cloud Consumer
+     *
+     * @throws UpstreamOrganizationNotExistException
+     *     Organization does not exist or could not be entitled
+     *
+     * @throws CouldNotEntitleOrganizationException
+     *     Organization could not be entitled
+     *
+     * @return key of the organization
+     */
+    String validateRegisteredOwner(String cloudAccountID, CloudProvider cloudProviderShortName,
+        Set<String> skuID) throws UpstreamOrganizationNotExistException, CouldNotEntitleOrganizationException;
```

Comment:
We had to change the design for this: instead of sending up the list of SKU IDs, we will be sending AnonConsumer.cloudOfferingID

Reply:
ack

Reply:
done